src/coreclr/tools/dotnet-pgo/Program.cs

```diff
@@ -354,6 +354,67 @@ static int InnerDumpMain(CommandLineOptions commandLineOptions)
             return 0;
         }
 
+        static MibcConfig ParseMibcConfig(TypeRefTypeSystem.TypeRefTypeSystemContext tsc, params PEReader[] pEReaders)
+        {
+            MibcConfig mergedConfig = new();
+            // When we merge multiple mibc let's just unconditionally pick the first valid MibcConfig
```

Comment:
Do we/Should we have any checks to verify that the .mibc files being combined are of the same .mibc config? Should one merge multiple mibc files of different config and we just choose the first one, then the merged .mibc will have data that is relevant for a different `runtime/os/arch`.

Reply:
> There are problems should one attempt to utilize a .mibc generated from a .nettrace obtained from a different Runtime. There are types in coreCLR that don't exist on mono, so mistakenly using a coreclr based .mibc with the Mono AOT Compiler causes problems.

I agree that we should emit a warning if one tries to merge profiles from conflicting runtimes. However, I think the runtimes still need to be hardened against invalid profiler data regardless -- it is bound to happen in end-user scenarios when profiles go stale.

Reply:
Added warnings.